airflow/api_fastapi/core_api/datamodels/dag_run.py

```diff
@@ -99,13 +100,15 @@ def check_data_intervals(cls, values):
     @model_validator(mode="after")
     def validate_dag_run_id(self):
         if not self.dag_run_id:
-            self.dag_run_id = DagRun.generate_run_id(DagRunType.MANUAL, self.logical_date)
+            self.dag_run_id = DagRun.generate_run_id(
+                run_type=DagRunType.MANUAL, logical_date=self.logical_date, run_after=self.run_after
+            )
         return self
 
     # Mypy issue https://github.com/python/mypy/issues/1362
     @computed_field  # type: ignore[misc]
     @property
-    def logical_date(self) -> datetime:
+    def run_after(self) -> datetime:
         return timezone.utcnow()
```

Comment:
I think this should be a field instead (can be specified by user, defaults to utcnow).

Reply:
Changed it